src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,27 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> eventListeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        eventListeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        eventListeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
```

Comment:
It's a good practice to handle the possibility of exceptions when submitting tasks to an ExecutorService. Consider adding error handling to ensure that any exceptions in the 'onEvent' method do not go unnoticed.